roles/kubernetes/preinstall/tasks/0120-growpart-azure-centos-7.yml

```
@@ -7,25 +7,34 @@
     name: cloud-utils-growpart
     state: present
 
+- name: Search root filesystem device
+  vars:
+    query: "[?mount=='/'].device"
+    _root_device: "{{ ansible_mounts|json_query(query) }}"
+  set_fact:
+    device: "{{ _root_device | first | regex_replace('([^0-9]+)[0-9]+', '\\1') }}"
+    partition: "{{ _root_device | first | regex_replace('[^0-9]+([0-9]+)', '\\1') }}"
```

```suggestion
    device: "{{ _root_device | regex_replace('([^0-9]+)[0-9]+', '\\1') }}"
    partition: "{{ _root_device | regex_replace('[^0-9]+([0-9]+)', '\\1') }}"
```
ah, if removing `| first ` from the above 2 parts, that would work fine:
```
    "ansible_facts": {
        "device": "/dev/sda",
        "partition": "1",
        "root_device": "/dev/sda1"
    },
```